driver-core/src/main/java/com/datastax/driver/core/DataType.java

```diff
@@ -726,10 +726,7 @@ public static ByteBuffer serializeValue(Object value, ProtocolVersion protocolVe
         try {
             return dt.serialize(value, protocolVersion);
         } catch (InvalidTypeException e) {
-            // In theory we couldn't get that if getDataTypeFor does his job correctly,
-            // but there is no point in sending an exception that the user won't expect if we're
-            // wrong on that.
-            throw new IllegalArgumentException(e.getMessage());
+            throw new IllegalArgumentException(e);
```

Comment:
We should also update the javadoc for this method to indicate that this method may also throw an exception if you try to serialize a value or type that isn't supported by the provided protocol version (as shown in the example in the PR description, another example:  Attempting to serialize a UDT with protocol version < 3).